services/director-v2/tests/unit/test_dynamic_sidecar_docker_api.py

```diff
@@ -0,0 +1,471 @@
+# pylint: disable=redefined-outer-name
+# pylint: disable=unused-argument
+
+import json
+from asyncio import BaseEventLoop
+from typing import Any, Dict, List
+from uuid import UUID, uuid4
+
+import pytest
+from models_library.projects import ProjectID
+from settings_library.docker_registry import RegistrySettings
+from simcore_service_director_v2.core.settings import DynamicSidecarSettings
+from simcore_service_director_v2.models.schemas.constants import (
+    DYNAMIC_PROXY_SERVICE_PREFIX,
+    DYNAMIC_SIDECAR_SERVICE_PREFIX,
+    UserID,
+)
+from simcore_service_director_v2.models.schemas.dynamic_services import (
+    ServiceLabelsStoredData,
+    ServiceState,
+    ServiceType,
+)
+from simcore_service_director_v2.modules.dynamic_sidecar.docker_api import (
+    are_all_services_present,
+    are_services_missing,
+    create_network,
+    create_service_and_get_id,
+    docker_client,
+    get_dynamic_sidecar_state,
+    get_dynamic_sidecars_to_monitor,
+    get_node_id_from_task_for_service,
+    get_swarm_network,
+    inspect_service,
+    is_dynamic_service_running,
+    list_dynamic_sidecar_services,
+    remove_dynamic_sidecar_network,
+    remove_dynamic_sidecar_stack,
+)
+from simcore_service_director_v2.modules.dynamic_sidecar.errors import (
+    DynamicSidecarError,
+    GenericDockerError,
+)
+
+pytestmark = pytest.mark.asyncio
+
+# FIXTURES
+
+
+@pytest.fixture(scope="module", autouse=True)
```

Comment:
what is a ``docker_swarm`` fixture doing in a unit test? this is not allowed. those can only be used if ``with_dbs`` OR ``integration tests``

WARNING: I find  ``autouse`` very difficult to debug and highly recommend to avoid using it

Reply:
I need to be in a docker swarm in oder to test the docker_api module which uses `aiodocker`.

Reply:
My reasons for placing these tests here is because they look more like unit tests to me and I'd like to keep them here. Also there are other unit tests which use aiodocker without mocking its API (they just don't require to be part of a swarm). That's how I ended up using that fixture.